harper-core/src/linting/phrase_corrections/mod.rs

```diff
@@ -58,10 +58,10 @@ pub fn lint_group() -> LintGroup {
             "Corrects `an` to `and` after `ahead`."
         ),
         "AllOfASudden" => (
-            ["all of the sudden", "all of sudden"],
-            ["all of a sudden"],
-            "The phrase is `all of a sudden`, meaning `unexpectedly`.",
-            "Corrects `all of the sudden` to `all of a sudden`.",
+            ["all of the sudden", "all of sudden", "all the sudden"],
+            ["all of a sudden", "all of the sudden"],
+            "Prefer the phrasing `all of a sudden` or `all of the sudden`.",
+            "Guides this expression toward `all of a sudden` or `all of the sudden`.",
```

Comment:
"All of the sudden" is nonstandard. I only know it from the internet. I think it's regional but not sure from where. Plenty of discussion about it vs standard "all of a sudden" on the internet:

- https://www.merriam-webster.com/grammar/usage-of-all-of-a-sudden
- https://www.grammarly.com/blog/commonly-confused-words/all-of-a-sudden/
- https://quillbot.com/blog/common-mistakes/all-of-a-sudden/
- https://www.rd.com/article/all-of-the-sudden-vs-all-of-a-sudden/

I would think we shouldn't recommend it. But if we must we definitely should present it is less standard than "all of a sudden".